activerecord/lib/active_record/insert_all.rb

```diff
@@ -181,12 +181,14 @@ def returning
           end
         end
 
-        def conflict_target
+        def conflict_target(connection_type = :postgresql)
```

Comment:
Hmmmm, I think it would if this method lived in the connection adapter. The general form could live in https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb, and then the Postgres adapter could override it as needed.

Reply:
Hey @ghiculescu! Thanks for the reply

I've just updated the solution with the given suggestion, migrating `conflict_target` to `abstract_adapter`, inherited from both adapters

Let me know if you have any additional suggestion

Reply:
Could you have a look at the failing tests?